# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/dialog/nice_scroll.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./menubox.c.orig	2003-03-09 12:09:44.000000000 +0100
+++ ./menubox.c	2003-03-09 12:24:36.000000000 +0100
@@ -259,6 +259,8 @@
 	scrollamt = choice - (max_choice - 1);
 	choice = max_choice - 1;
     }
+    while ( (scrollamt+max_choice < item_no) &&
+	(choice > max_choice/2) ) { scrollamt++; choice--; }
 
     /* Print the menu */
     for (i = 0; i < max_choice; i++)
